core/node/eth_watch/src/client.rs

```diff
@@ -616,19 +652,19 @@ impl L2EthClient for EthHttpQueryClient<L2> {
     }
 }
 
-/// Wrapper for L2 client object.
-/// It is used for L2EthClient -> EthClient dyn upcasting coercion:
-///     Arc<dyn L2EthClient> -> L2EthClientW -> Arc<dyn EthClient>
+/// Wrapper for ZkSyncExtention client object.
+/// It is used for ZkSyncExtentionEthClient -> EthClient dyn upcasting coercion:
+///     Arc<dyn ZkSyncExtentionEthClient> -> ZkSyncExtentionEthClientW -> Arc<dyn EthClient>
 #[derive(Debug, Clone)]
-pub struct L2EthClientW(pub Arc<dyn L2EthClient>);
+pub struct ZkSyncExtentionEthClientW(pub Arc<dyn ZkSyncExtentionEthClient>);
```

Comment:
This entire construction is unnecessary. You can add the upcasting method directly into the `ZkSyncExtentionEthClient` trait:

```rust
#[async_trait::async_trait]
pub trait ZkSyncExtentionEthClient: EthClient {
    fn into_base(self: Arc<Self>) -> Arc<dyn EthClient>;
    // other stuff...
}

#[async_trait::async_trait]
impl ZkSyncExtentionEthClient for EthHttpQueryClient<L1> {
    fn into_base(self: Arc<Self>) -> Arc<dyn EthClient> {
        self
    }
    // ...
}
```

Unfortunately, upcasting cannot have a default implementation, but it's trivial to add it to the 3 impls.